rct229/rules/section5/section5rule17.py

```diff
@@ -0,0 +1,158 @@
+        def __init__(self):
+            super(Section5Rule17.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, True),
+                required_fields={},
+                each_rule=Section5Rule17.BuildingRule.UnregulatedSurfaceRule(),
+                index_rmr="baseline",
+            )
+
+        def create_context_list(self, context, data=None):
+            # List of all baseline roof surfaces to become the context for RoofRule
+            scc = data["surface_conditioning_category_dict"]
+
+            baseline_surfaces = find_all("$..surfaces[*]", context.baseline)
+            proposed_surfaces = find_all("$..surfaces[*]", context.proposed)
+
+            # This assumes that the surfaces matched by IDs between proposed and baseline
+            matched_proposed_surfaces = match_lists_exactly_by_id(
+                baseline_surfaces, proposed_surfaces
+            )
+
+            proposed_baseline_surface_pairs = zip(baseline_surfaces, matched_proposed_surfaces)
+
+            return [
+                UserBaselineProposedVals(None, surface_b, surface_p)
+                for surface_b, surface_p in proposed_baseline_surface_pairs
+                if scc[surface_b["id"]] == SCC.UNREGULATED
+            ]
+
+        def create_data(self, context, data=None):
+            building = context.baseline
+            # Merge into the existing data dict
+            return {
+                **data,
+                "surface_conditioning_category_dict": get_surface_conditioning_category_dict(
+                    data["climate_zone"], building
+                ),
+            }
+
+        class UnregulatedSurfaceRule(RuleDefinitionBase):
+            def __init__(self):
+                super(Section5Rule17.BuildingRule.UnregulatedSurfaceRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, True),
+                    required_fields={"$": ["construction"]},
+                )
+
+            def get_calc_vals(self, context, data=None):
+                surface_b = context.baseline
+                surface_p = context.proposed
+
+                surface_b_type = get_opaque_surface_type(surface_b)
+                surface_b_construction = surface_b["construction"]
+                surface_p_type = None
+                surface_p_construction = None
+
+                if surface_p is not None:
+                    surface_p_type = get_opaque_surface_type(surface_p)
+                    surface_p_construction = surface_p["construction"]
+
+                if surface_b_type in [OST.ABOVE_GRADE_WALL, OST.FLOOR, OST.ROOF]:
+                    return {
+                        "id": surface_b["id"],
+                        "baseline_surface_type": surface_b_type,
+                        "proposed_surface_type": surface_p_type,
+                        "baseline_surface_u_factor": surface_b_construction.get("u_factor"),
```

Comment:
In this case, should we require the u_factor field?
```
"baseline_surface_u_factor": getattr_(surface_b_construction, "surface_b_construction", "u_factor")
```
Otherwise, how do we handle the case that both baseline_surface_u_factor and proposed_surface_u_factor are None?  Is that passing or failing?  Currently, if neither is specified in the RMD, it passes.

Reply:
Yes, that is one question I have. 
My understanding is that if both are None, then the rule test fails. I can bring this up in the next RDS meeting to clarify the issue.

I prefer to raise the exception flag in the rule_check function, where it handles the outcome of the rule testing. thoughts?